### PR TITLE
refactor(atoms): SfButton

### PR DESCRIPTION
**Pull request by aniamusial**
[IMPORTANT CHANGE] 
- added modifier for underlined: https://www.figma.com/file/N0Ct95cSAoODNv7zYS01ng/Storefront-UI-Design-System?node-id=1%3A18
- added slot for additional icon next to button text

# Checklist

- [x] No commented blocks of code left
- [x] Run tests and docs
- [x] Self code-reviewed

### CODE REVIEW

File: packages/shared/styles/components/atoms/SfButton.scss [hunk 1]
```diff
@@ -4,104 +4,154 @@
...
   height: var(--button-size, var(--button-height));
-  padding: var(--button-padding, var(--spacer-medium) var(--spacer-extra-big));
+  display: flex;
+  justify-content: center;
+  padding: var(
+    --button-padding,
+    var(--spacer-small) calc(var(--spacer-small) * 2)
+  );
+  color: var(--button-color, var(--c-light-variant));
   border: var(--button-border, 0);
```
Comment (martaradziszewska) [inline packages/shared/styles/components/atoms/SfButton.scss]: Here, it's better to use mixin `@include border()` :)

File: packages/shared/styles/components/atoms/SfButton.scss [hunk 2]
```diff
@@ -4,104 +4,154 @@
...
+          &:hover {
+            --button-color: var(--c-text-muted);
+          }
+        }
       }
     }
-    &:disabled {
+  }
+
+  &--underlined {
```
Comment (pspaczek) [inline packages/shared/styles/components/atoms/SfButton.scss]: @aniamusial  IMO we should use it for desktop too

File: packages/shared/styles/components/atoms/SfButton.scss [hunk 3]
```diff
@@ -4,104 +4,154 @@
...
-        --button-background: var(--c-#{$color}-lighten);
+      &:active {
+        --button-background: #{$variant};
       }
     }
   }
+
+  &:hover {
+    --button-box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+    --button-transition: box-shadow 150ms ease-in-out;
```
[the remark this answers is not in the document]
Comment (pspaczek) [reply packages/shared/styles/components/atoms/SfButton.scss]: --button-transition: box-shadow 150ms ease-in-out;

File: packages/shared/styles/components/atoms/SfButton.scss [hunk 4]
```diff
@@ -4,104 +4,152 @@
...
-      --button-border: 1px solid var(--c-gray);
-      --button-background: transparent;
-      --button-color: var(--c-gray);
-    }
   }
+
   &--text {
-    --button-border: 0;
+    --button-border-width: 0;
     --button-padding: var(--spacer-small) 0;
```
Comment (pspaczek) [inline packages/shared/styles/components/atoms/SfButton.scss]:
```suggestion
    --button-padding: 0;
```
@aniamusial for `--text` we can set padding: 0

**Pull request merged**